src/api/schemas/user-schema.ts

```diff
@@ -0,0 +1,57 @@
+import { object, string } from 'zod';
+
+const createUserSchema = object({
+    body: object({
+        firstName: string({
+            required_error: 'Firstname is required.'
+        }).min(2, 'First name too short!'),
+        lastName: string({
+            required_error: 'Lastname is required.'
+        }).min(2, 'Last name too short!'),
+        email: string({
+            required_error: 'Email is required.'
+        }).email('Not a valid email.'),
+        countryCode: string({
+            required_error: 'Country Code is required.'
+        }).regex(new RegExp('^\\d\\d?$'), 'Invalid country code.'),
+        phone: string({
+            required_error: 'Phone number is required.'
+        }).regex(new RegExp('^\\d{10}$'), 'Please enter a valid phone number (10 digits).')
+    })
+});
+
+const getUserByEmailSchema = object({
+    body: object({
+        email: string({
+            required_error: 'Email is required.'
+        }).email('Not a valid email.')
+    })
+});
+
+const updateUserSchema = object({
+    body: object({
```

Comment:
When I use postman to update a user without any json body, it returns an incomplete error message:
![image](https://user-images.githubusercontent.com/64218770/195694902-a9fc4fb6-0b30-4035-9b56-60c896258c2a.png)

Reply:
Yeah, absolutely right. I failed to find an ideal solution within a reasonable amount of time. I'm trying again today. Otherwise I will just add the validation in the controller.